logical/src/main/java/org/apache/drill/common/logical/PlanProperties.java

```diff
@@ -112,8 +121,13 @@ public PlanPropertiesBuilder generator(Generator generator) {
       return this;
     }
 
+    public PlanPropertiesBuilder generator(boolean hasResourcePlan) {
```

Comment:
The method's name should not be **generator** but something about having a resource plan